kubectl-cloudflow/cloudflowavro/avrocompatibility.go

```diff
@@ -0,0 +1,48 @@
+package cloudflowavro
+
+import (
+	"github.com/hamba/avro"
+	"github.com/linkedin/goavro/v2"
+)
+
+func ValidateSchema(schema string) (*string, error) {
+	codec, err := goavro.NewCodec(schema)
+	if err != nil {
+		return nil, err
+	} else {
+		canonical := codec.CanonicalSchema()
+		return &canonical, nil
+	}
+}
+
+type ReaderSchemaValidationError struct {
+	msg string
+}
+
+type WriterSchemaValidationError struct {
+	msg string
+}
+
+func (e *ReaderSchemaValidationError) Error() string { return e.msg }
+
+func (e *WriterSchemaValidationError) Error() string { return e.msg }
+
+func CheckSchemaCompatibility(reader string, writer string) error {
```

Comment:
Is there a `Schema` type of sorts in the avro package we could use instead of strings?

Reply:
As we discussed this is not possible at the moment so have to use two different libraries here.

Reply:
yes agree